### PR TITLE
Fix default data_format for Conv1D

### PR DESCRIPTION
**Pull request by yanboliang**
### Summary
The default ```data_format``` for ```Conv1D``` should be decided by Keras config at ```~/.keras/keras.json```, so it should be ```None``` in the constructor. I also checked all other convolution layers, they are correctly set with ```None```.

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

File: tests/keras/layers/convolutional_test.py [hunk 1]
```diff
@@ -24,15 +24,17 @@
     [
         # Causal
         ({'filters': 1, 'kernel_size': 2, 'dilation_rate': 1, 'padding': 'causal',
-          'kernel_initializer': 'ones', 'use_bias': False},
+          'kernel_initializer': 'ones', 'data_format': 'channels_last',
+          'use_bias': False},
```
Comment (yanboliang) [inline tests/keras/layers/convolutional_test.py]: Causal convolution only supports ```channels_last```

**Comment by fchollet** [comment]
That's incorrect; a conv1d is a temporal convolution and it should default to `channels_last`, because by default the time axis is 1.
